federation-js/src/service/printCoreSchema.ts

```diff
@@ -278,14 +278,22 @@ function printEnum(type: GraphQLEnumType, options?: Options): string {
         printDescription(options, value, '  ', !i) +
         '  ' +
         value.name +
-        printDeprecated(value),
+        printDeprecated(value) +
+        printHttpDirective(type, value),
     );
 
   return (
     printDescription(options, type) + `enum ${type.name}` + printBlock(values)
   );
 }
 
+function printHttpDirective(type: GraphQLEnumType, value: GraphQLEnumValue) {
+  if (type.name === "join__Graph") {
+    return ` @http(url: "${value.value}")`
+  }
+  return '';
+}
+
```

Comment:
I renamed this to `join__endpoint` and added a `serviceName` argument. We also need to define the directive and include it in the SDL (see a09e4339b6232ce400a40edc9e0694b681f4b7a2).

Reply:
`cherry-pick`ed a09e433